### PR TITLE
WIP: llvm: update to 7.0.0.

### PR DESCRIPTION
**Pull request by Cogitri**
Thanks to @maxice8, who did most of the actual work of bumping LLVM.

### CODE REVIEW

**Comment by Cogitri** [comment]
Tested&Builds:
- [X] libGL
- [X] beignet
- [ ] crystal
- [ ] ispc
- [ ] julia
- [ ] qtcreator
- [ ] codelite
- [ ] gnome-builder
- [ ] kdevelop
- [ ] rtags

**Comment by Cogitri** [comment]
Seems like crystal doesn't support LLVM 7 yet, opened an issue upstream: https://github.com/crystal-lang/crystal/issues/6754

**Comment by maxice8** [comment]
please add [ci skip]

**Comment by Cogitri** [comment]
Breaks rust  1.28.x right now.

**Comment by archfan** [comment]
Any new developments?

**Comment by maxice8** [comment]
On Thu, Nov 29, 2018 at 02:09:26PM -0800, archfan wrote:
> Any new developments?
> 
> -- 
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/2918#issuecomment-443012195

Waiting new Crystal-lang release.

**Comment by RX14** [comment]
crystal isn't likely to upgrade to llvm7 soon, nobody is working on it. I suggest you do like arch and make llvm6 packages

**Comment by Cogitri** [comment]
Currently testing that everything works with both llvm7 and that crystal works with llvm6...

File: common/shlibs [hunk 1]
```diff
@@ -1001,10 +1001,11 @@ libuchardet.so.0 uchardet-0.0.6_1
 libgtksourceviewmm-3.0.so.0 gtksourceviewmm-3.2.0_1
 libyajl.so.2 yajl-2.0.1_1
 libconfuse.so.2 confuse-3.2.1_1
-liblldb.so.6 lldb-6.0.0_1
-libclang.so.6 clang-6.0.0_1
+liblldb.so.7 lldb-7.0.0_1
+libclang.so.7 clang-7.0.0_1
 libLLVM-3.9.so libllvm3.9-3.9.0_1
 libLLVM-6.0.so libllvm6.0-6.0.0_1
+libLLVM-7.so libllvm7.0-7.0.0_1
```
Comment (maxice8) [inline common/shlibs]: libllvm7 since that is the soname

**Comment by Cogitri** [comment]
Moved llvm7 -> llvm, rebuilding...

File: srcpkgs/libcxx/template [hunk 2]
```diff
@@ -1,53 +1,52 @@
...
+version=7.0.0
 revision=1
 wrksrc="${pkgname}-${version}.src"
 build_style=cmake
-hostmakedepends="clang llvm python"
+configure_args+=" -DLIBCXX_CXX_ABI=libcxxabi -DLIBCXX_ENABLE_STATIC_ABI_LIBRARY=ON
+ -DLIBCXX_CXX_ABI_LIBRARY_PATH=${XBPS_CROSS_BASE}/usr/lib
+ -DLIBCXXABI_USE_LLVM_UNWINDER=1"
+hostmakedepends="llvm python"
 makedepends="libcxxabi-devel llvm llvm-libunwind-devel"
```
Comment (maxice8) [inline srcpkgs/libcxx/template]: missing here

**Comment by q66** [comment]
hi,

i'd like to request that this patch is included https://code.foxkit.us/adelie/packages/blob/master/user/llvm6/musl-ppc64-elfv2.patch in the llvm build

this will make it easier to deal with llvm when the ppc64 support is merged, as this affects cross-compiling to ppc64-musl besides native ppc64-musl builds, so otherwise I would have to add the patch later and bump revision, forcing a rebuild of llvm everywhere again - this should leave the llvm prepared.

thanks!

**Comment by Cogitri** [comment]
More stuff to be tested:
- [ ] rust
- [ ] llvm-libunwind
- [ ] thunderbird
- [ ] ldc
- [ ] icecat
- [ ] firefox
- [ ] clazy
- [ ] blender
- [ ] bcc
- [ ] afl

All other stuff works AFAICS.

**Comment by Cogitri** [comment]
I've added the patch @q66, will squash once I've tested everything.

**Comment by Cogitri** [comment]
7.0.1 has been released, but upstream will release 7.1.x or 7.0.2 soon due to https://bugs.llvm.org/show_bug.cgi?id=39427

**Comment by Cogitri** [comment]
Or we could apply that patch here

**Comment by Cogitri** [comment]
Alright, from what I can see this should work now. Please test this, binaries for x86_64 are available on https://exp.exqa.de

**Comment by maxice8** [comment]
merged

**Pull request closed without merging**